### PR TITLE
#340 use transaction for manual capturing

### PR DESCRIPTION
**Pull request by lojzatran**
Fixes: https://github.com/commercetools/commercetools-adyen-integration/issues/340

### CODE REVIEW

File: extension/src/paymentHandler/manual-capture.handler.js [hunk 1]
```diff
@@ -2,26 +2,22 @@ const pU = require('./payment-utils')
 const { manualCapture } = require('../service/web-component-service')
 const {
   CTP_INTERACTION_TYPE_MANUAL_CAPTURE,
-  CTP_CUSTOM_FIELD_MANUAL_CAPTURE_RESPONSE
```
Comment (ahmetoz) [inline extension/src/paymentHandler/manual-capture.handler.js:5]: please remove unused constants.

File: extension/docs/ManualCapture.md [hunk 2]
```diff
@@ -52,54 +37,56 @@ Additionally, a [Charge](https://docs.commercetools.com/http-api-projects-paymen
 }
```
Comment (ahmetoz) [inline extension/docs/ManualCapture.md:37]: interaction id should be 883592922122044A, right ?

Comment (lojzatran) [reply extension/docs/ManualCapture.md:37]: https://github.com/commercetools/commercetools-adyen-integration/pull/341/commits/a71be09ccf9cae86dd9a737db2b2314ac4b76377

File: extension/src/paymentHandler/manual-capture.handler.js (hunk 1 above)
Comment (lojzatran) [reply extension/src/paymentHandler/manual-capture.handler.js:5]: https://github.com/commercetools/commercetools-adyen-integration/pull/341/commits/846974cdb794c986884ba4880c2b1ef1c60c0906

**Comment by lojzatran** [comment]
> please update :
> 
> * payment type
> * json validations
> * constants.

https://github.com/commercetools/commercetools-adyen-integration/pull/341/commits/846974cdb794c986884ba4880c2b1ef1c60c0906

File: extension/src/paymentHandler/manual-capture.handler.js [hunk 3]
```diff
@@ -2,26 +2,22 @@ const pU = require('./payment-utils')
...
-      type: 'Charge',
-      state: 'Pending',
-      amount: manualCaptureRequestObj.modificationAmount.value,
-      currency: manualCaptureRequestObj.modificationAmount.currency,
-      interactionId: response.pspReference
-    }))
+  if (!response.errorCode && response.pspReference) {
+    actions.push(pU.createChangeTransactionStateAction(chargeInitialTransaction.id, 'Pending'))
+    actions.push(pU.createChangeTransactionInteractionId(chargeInitialTransaction.id, response.pspReference))
+  }
```
Comment (ahmetoz) [inline extension/src/paymentHandler/manual-capture.handler.js:21]: can you remove those extra empty lines, in general, can you check the format?

File: extension/src/service/web-component-service.js [hunk 4]
```diff
@@ -24,7 +24,14 @@ function cancelOrRefund (cancelOrRefundRequestObj) {
   return callAdyen(`${config.adyen.legacyApiBaseUrl}/cancelOrRefund`, cancelOrRefundRequestObj)
 }
 
-function manualCapture (manualCaptureRequestObj) {
+function manualCapture (chargePendingTransaction, authorizationTransactionInteractionId) {
+  const manualCaptureRequestObj = {
```
Comment (ahmetoz) [inline extension/src/service/web-component-service.js:28]: Could you please put those lines into `execute` method to be able to follow to contract like other Adyen calls?

File: extension/src/paymentHandler/manual-capture.handler.js [hunk 5]
```diff
@@ -2,26 +2,22 @@ const pU = require('./payment-utils')
...
-  const manualCaptureRequestObj = JSON.parse(paymentObject.custom.fields.manualCaptureRequest)
-  const { request, response } = await manualCapture(manualCaptureRequestObj)
+  const chargeInitialTransaction = pU.getChargeTransactionInitial(paymentObject)
+  const authorizationSuccessTransaction = pU.getAuthorizationTransactionSuccess(paymentObject)
+  const { request, response } = await manualCapture(chargeInitialTransaction,
+    authorizationSuccessTransaction.interactionId)
   const actions = [
     pU.createAddInterfaceInteractionAction({
       request, response, type: CTP_INTERACTION_TYPE_MANUAL_CAPTURE
     }),
```
Comment (ahmetoz) [inline extension/src/paymentHandler/manual-capture.handler.js:15]: could you please check the formatting of the execute method, for example in here we could remove comma,
and use spaces better for between lines to improve readability.

File: extension/src/paymentHandler/manual-capture.handler.js [hunk 6]
```diff
@@ -2,26 +2,22 @@ const pU = require('./payment-utils')
...
   CTP_INTERACTION_TYPE_MANUAL_CAPTURE,
-  CTP_CUSTOM_FIELD_MANUAL_CAPTURE_RESPONSE
 } = require('../config/constants')
 
 async function execute (paymentObject) {
-  const manualCaptureRequestObj = JSON.parse(paymentObject.custom.fields.manualCaptureRequest)
-  const { request, response } = await manualCapture(manualCaptureRequestObj)
+  const chargeInitialTransaction = pU.getChargeTransactionInitial(paymentObject)
+  const authorizationSuccessTransaction = pU.getAuthorizationTransactionSuccess(paymentObject)
+  const { request, response } = await manualCapture(chargeInitialTransaction,
```
Comment (ahmetoz) [inline extension/src/paymentHandler/manual-capture.handler.js:10]: actually interaction id for add transaction is optional, maybe we should not add it in the first place, WDYT?

Comment (lojzatran) [reply extension/src/paymentHandler/manual-capture.handler.js:10]: The interaction ID is quite useful. It helps you to connect with the interfaceInteraction. So I would add it there.

Comment (ahmetoz) [reply extension/src/paymentHandler/manual-capture.handler.js]: OK, I check again, it's different than what I thought.

File: extension/src/paymentHandler/manual-capture.handler.js (hunk 3 above)
Comment (lojzatran) [reply extension/src/paymentHandler/manual-capture.handler.js:29]: https://github.com/commercetools/commercetools-adyen-integration/pull/341/commits/87dd9e28d8b0b70a5ee1bc232f52f2b12cb4ce9f

File: extension/src/service/web-component-service.js (hunk 4 above)
Comment (lojzatran) [reply extension/src/service/web-component-service.js]: https://github.com/commercetools/commercetools-adyen-integration/pull/341/commits/87dd9e28d8b0b70a5ee1bc232f52f2b12cb4ce9f

File: extension/src/paymentHandler/manual-capture.handler.js (hunk 5 above)
Comment (lojzatran) [reply extension/src/paymentHandler/manual-capture.handler.js]: https://github.com/commercetools/commercetools-adyen-integration/pull/341/commits/87dd9e28d8b0b70a5ee1bc232f52f2b12cb4ce9f

**Pull request merged**